domain/ide/workspace/workspace.go

```diff
@@ -174,18 +174,17 @@ func (w *Workspace) ScanWorkspace(ctx context.Context) {
 
 // ChangeWorkspaceFolders clears the "Removed" folders, adds the "New" folders,
 // and starts an automatic scan if auto-scans are enabled.
-func (w *Workspace) ChangeWorkspaceFolders(ctx context.Context, params types.DidChangeWorkspaceFoldersParams) {
+func (w *Workspace) ChangeWorkspaceFolders(params types.DidChangeWorkspaceFoldersParams) []*Folder {
 	for _, folder := range params.Event.Removed {
 		w.RemoveFolder(uri.PathFromUri(folder.Uri))
 	}
-
+	var changedWorkspaceFolders []*Folder
```

Comment:
this is to avoid triggering scan before initializing the persister.
Caller is responsible for calling the scan for the new added folders